django_advanced_queries/covid_19/tests.py

```diff
@@ -457,20 +499,55 @@ def test_number_of_hospital_workers_that_in_risk_group_of_corona_per_hospital(se
     def test_annotate_by_num_of_dead_from_corona(self):
         # Dead from corona is someone who had corona and then died
         with self.assertNumQueries(1):
-            result = Hospital.objects.\
-                annotate_by_num_of_dead_from_corona().order_by()
-
-            hospital1_num_of_dead_from_corona = result[0].num_of_dead_from_corona
+            examination_results = MedicalExaminationResult.objects.filter(
+                patient=OuterRef('pk')).order_by('-time').values('result')
+
+            patient_died_from_corona = Patient.objects.annotate(
+                last_result=Subquery(
+                    examination_results[:1]),
+                second_last_result=Subquery(examination_results[1:2])).filter(
+                department__hospital=OuterRef('pk'),
+                last_result=MedicalExaminationResult.RESULT_DEAD,
+                second_last_result=MedicalExaminationResult.RESULT_CORONA)
+
+            result = list(Hospital.objects.annotate(
+                num_of_dead_from_corona=Count(Case(When(Q(
+                    departments__patients_details__in=Subquery(
+                        patient_died_from_corona.values('pk'))),
+                    then=F('departments__patients_details__person'))),
+                    distinct=True)).order_by())
+
+            hospital1_num_of_dead_from_corona = result[
+                0].num_of_dead_from_corona
             self.assertEqual(hospital1_num_of_dead_from_corona, 0)
 
-            hospital2_num_of_dead_from_corona = result[1].num_of_dead_from_corona
+            hospital2_num_of_dead_from_corona = result[
+                1].num_of_dead_from_corona
             self.assertEqual(hospital2_num_of_dead_from_corona, 2)
 
     def test_hospitals_with_at_least_two_dead_patients_from_corona(self):
         # Dead from corona is someone who had corona and then died
         with self.assertNumQueries(1):
             # Define query by yourself
-            hospitals_with_more_than_two_dead_patients_from_corona = None
+            examination_results = MedicalExaminationResult.objects.filter(
```

Comment:
Would be better to implement this logic as part of query manager so it can be reused